hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastConfigBeanDefinitionParser.java

```diff
@@ -335,6 +339,18 @@ private AbstractBeanDefinition handleService(Node node) {
                     }
                 }
             }
+
+            final NamedNodeMap attrs = node.getAttributes();
+            Node classNameNode = attrs.getNamedItem("class-name");
+            if (classNameNode != null) { 
+            	serviceConfigBuilder.addPropertyValue("className", getTextContent(classNameNode));
+            }
+            Node implNode = attrs.getNamedItem("implementation");
+            if (implNode != null) { 
+                serviceConfigBuilder.addPropertyReference("implementation", getTextContent(implNode));
+            }
+            Assert.isTrue(classNameSet || classNameNode != null || implNode != null, "One of 'class-name' or 'implementation' "
```

Comment:
why do you think it is wrong? it s very similar to all other places where pair of class-name/implementation attributes is processed. the only difference is that I check for null attributes itself and not their string values. but they can not be empty by their type specified in XSD: non-space-string.